### PR TITLE
[INLONG-4232][Agent][Manager] Support collect data from a specified position for MySQL binlog 

### LINKED ISSUE #4232: [Feature][Agent][Manager] Support collect data from a specified position for MySQL binlog
### Description

mysql binlog supports collection from a specified location

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Greedyu**
Fixes #4232
1. Mangger support collect data from a specified position for MySQL binlog.
2. Manager-client-tool supports creating groups.

### CODE REVIEW

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/source/MySQLBinlogSource.java [hunk 1]
```diff
@@ -100,4 +100,10 @@ public class MySQLBinlogSource extends StreamSource {
 
     @ApiModelProperty(value = "Primary key must be shared by all tables", required = false)
     private String primaryKey;
+
+    @ApiModelProperty("Directly read binlog from the specified offset filename")
+    public String specificOffsetFile;
```
Comment (healchow) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/source/MySQLBinlogSource.java:105]: Maybe change to the `offsetFile` and `offsetPosition` are more readable.

File: inlong-manager/manager-client-tools/src/main/java/org/apache/inlong/manager/client/cli/CommandCreate.java [hunk 2]
```diff
@@ -71,7 +71,7 @@ void run() {
                 }
             } catch (Exception e) {
                 System.out.println("Create group failed!");
-                System.out.println(e.getMessage());
```
Comment (healchow) [inline inlong-manager/manager-client-tools/src/main/java/org/apache/inlong/manager/client/cli/CommandCreate.java:74]: Just printing the message of the exception is enough, thanks.

File: inlong-manager/manager-client-tools/src/test/resources/create_group.json [hunk 3]
```diff
@@ -0,0 +1,98 @@
+{
+  "groupInfo": {
+    "groupName": "test_group14",
+    "inlongGroupId": "b_test_group45",
+    "description": "",
+    "inCharges": "admin",
+    "proxyClusterId": "2",
+    "mqType": "PULSAR",
+    "mqBaseConf": {
```
Comment (healchow) [inline inlong-manager/manager-client-tools/src/test/resources/create_group.json:9]: That MQ config was changed, please change it with the latest branch, thanks.

**Review by healchow — CHANGES_REQUESTED** [review]

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/source/MySQLBinlogSource.java (hunk 1 above)
Comment (Greedyu) [reply inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/source/MySQLBinlogSource.java:105]: Referring to the names used by Flink CDC are specificOffsetFile and specificOffsetPos, and the agent module also uses this name.
image

File: inlong-manager/manager-client-tools/src/main/java/org/apache/inlong/manager/client/cli/CommandCreate.java (hunk 2 above)
Comment (Greedyu) [reply inlong-manager/manager-client-tools/src/main/java/org/apache/inlong/manager/client/cli/CommandCreate.java:74]: The main reason is that the e.getMessage() method cannot see the abnormal link, and the abnormal information is not enough.

File: inlong-manager/manager-client-tools/src/main/java/org/apache/inlong/manager/client/cli/CommandCreate.java [hunk 4]
```diff
@@ -71,7 +71,7 @@ void run() {
...
+                e.printStackTrace();
```
Comment (EMsnap) [inline inlong-manager/manager-client-tools/src/main/java/org/apache/inlong/manager/client/cli/CommandCreate.java:74]: printing stack trace is not proper

File: inlong-manager/manager-client-tools/src/test/resources/create_group.json (hunk 3 above)
Comment (Greedyu) [reply inlong-manager/manager-client-tools/src/test/resources/create_group.json]: resolve

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/impl/DefaultInlongStreamBuilder.java [hunk 5]
```diff
@@ -204,24 +204,26 @@ private void initOrUpdateSource() {
...
-                    throw new RuntimeException(String.format("Delete source=%s failed", sourceListResponse));
-                }
-            } else {
-                SourceRequest sourceRequest = sourceRequests.get(sourceName);
-                sourceRequest.setId(id);
-                Pair<Boolean, String> updateState = managerClient.updateSource(sourceRequest);
-                if (!updateState.getKey()) {
-                    throw new RuntimeException(String.format("Update source=%s failed with err=%s", sourceRequest,
-                            updateState.getValue()));
+        if (sourceListResponses != null && sourceListResponses.size() > 0) {
```
Comment (healchow) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/impl/DefaultInlongStreamBuilder.java:207]: Please use the `CollectionUtils.isNotEmpty` method to check whether the list is empty or not.

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/impl/DefaultInlongStreamBuilder.java [hunk 6]
```diff
@@ -204,24 +204,26 @@ private void initOrUpdateSource() {
...
+            for (SourceListResponse sourceListResponse : sourceListResponses) {
+                final String sourceName = sourceListResponse.getSourceName();
+                final int id = sourceListResponse.getId();
+                if (sourceRequests.get(sourceName) == null) {
+                    boolean isDelete = managerClient.deleteSource(id);
+                    if (!isDelete) {
+                        throw new RuntimeException(String.format("Delete source=%s failed", sourceListResponse));
```
Comment (healchow) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/impl/DefaultInlongStreamBuilder.java:214]: Suggested change the exception message to `throw new RuntimeException(String.format("Delete source failed by id=%s", id));`, thanks.

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/sort/BaseSortConf.java [hunk 7]
```diff
@@ -25,14 +27,21 @@
  */
 @Data
 @ApiModel("Sort configuration for inlong group")
+@JsonTypeInfo(use = Id.NAME, visible = true, property = "sortTypeString")
 public abstract class BaseSortConf {
 
+    public String sortTypeString;
```
Comment (healchow) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/sort/BaseSortConf.java:33]: Suggested change to `sortType`.

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/impl/DefaultInlongStreamBuilder.java (hunk 5 above)
Comment (Greedyu) [reply inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/impl/DefaultInlongStreamBuilder.java]: resolve

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/impl/DefaultInlongStreamBuilder.java (hunk 6 above)
Comment (Greedyu) [reply inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/impl/DefaultInlongStreamBuilder.java]: resolve

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/sort/BaseSortConf.java (hunk 7 above)
Comment (Greedyu) [reply inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/sort/BaseSortConf.java]: resolve

File: inlong-manager/manager-client-tools/src/main/java/org/apache/inlong/manager/client/cli/CommandCreate.java (hunk 4 above)
Comment (Greedyu) [reply inlong-manager/manager-client-tools/src/main/java/org/apache/inlong/manager/client/cli/CommandCreate.java]: Modified to Logger output

**Review by healchow — APPROVED** [review]
LGTM

**Review by EMsnap — APPROVED** [review]

**Review by pocozh — APPROVED** [review]
+1

**Pull request merged**